server/base/src/main/java/org/apache/accumulo/server/rpc/CustomNonBlockingServer.java

```diff
@@ -112,20 +112,20 @@ protected FrameBuffer createFrameBuffer(final TNonblockingTransport trans,
    * extract the client's network location before accepting the request.
    */
   private class CustomFrameBuffer extends FrameBuffer {
+    private final String clientAddress;
 
     public CustomFrameBuffer(TNonblockingTransport trans, SelectionKey selectionKey,
         AbstractSelectThread selectThread) throws TTransportException {
       super(trans, selectionKey, selectThread);
+      // Store the clientAddress in the buffer so it can be referenced for logging during read/write
+      this.clientAddress = getClientAddress();
     }
 
     @Override
     public void invoke() {
-      if (trans_ instanceof TNonblockingSocket) {
-        TNonblockingSocket tsock = (TNonblockingSocket) trans_;
-        Socket sock = tsock.getSocketChannel().socket();
-        TServerUtils.clientAddress
-            .set(sock.getInetAddress().getHostAddress() + ":" + sock.getPort());
-      }
+      // On invoke() set the clientAddress on the ThreadLocal so that it can be accessed elsewhere
+      // in the same thread that called invoke() on the buffer
+      TServerUtils.clientAddress.set(clientAddress);
```

Comment:
This preserves current behavior. I don't know if that current behavior works all that well, but it should suffice for now, in order to limit the scope of this PR.

Reply:
Agree that the current behavior may not actually be best but it at least accomplishes the goal to fix the logging without potentially causing problems. For what it's worth, it seems like the current behavior has existed for a long time going back several years so it is well tested. I do think ultimately I'd like to get changes into the Thrift api at some point as you had mentioned so we don't have to actually store the value in this way.